### ISSUE TITLE
`vagrant plugin install --plugin-source` does not use defined source

### ISSUE THREAD

**Issue by LongLiveCHIEF** [mitchellh/vagrant#8287]
### Vagrant version

1.9.1

### Host operating system

Tested on Mac OSx Sierra, Windows 7, Windows 10

### Guest operating system

N/A, no guests used, this is a ticket for plugin management

### Vagrantfile
N/A

### Debug output
N/A

### Expected behavior

I expected the defined source to be the _only_, or at least the _primary_ source used when I ran:

```
$ vagrant plugin install --plugin-source https://private.rubygems.mirror.org vagrant-vbguest
```

### Actual behavior

The plugin was installed from rubygems.org, even though I specified a different source. I inspected the `~/.vagrant.d/plugins.json` file, and found that each installed plugin added the `rubygems.org` and `gems.hashicorp.com` in addition to my defined plugin source, and it added them _before_ my specified source.

### Steps to reproduce
1. setup a Nexus Repository Manager locally using a the [docker image](https://github.com/sonatype/docker-nexus), (make sure to do a host port map to `8081`
2.  Enable the [rubygems.org proxy mirror](http://books.sonatype.com/nexus-book/reference3/rubygems.html#rubygems-proxy) repository
3. run `vagrant plugin install --plugin-source http://localhost:8081/repository/rubygems.org vagrant-vbguest` -> plugin will install successfully
4. Browse to your nexus component library, using `http://localhost:8081/#browse/browse/components:rubygems.org` and see it will be empty 

I reviewed the plugins.json file in
*note*: I _do_ have proxy environment variables defined.

### References

Was unable to find any tickets relating specifically on disabling plugin sources, or defining global sources.

### Requested Bug Fix

If a `--plugin-source` is defined, it should be at the very least, the _first_ gem source attempted to install from, if not the _only_ source.

### Requested Enhancement

The `plugins.json` file should respect a top-level `sources` object, with the following specs:

- a `sources` key of type `Array`, which take an array of source object literals
- a `source` object has two keys: `source` (value of url string), and `enabled` (value of boolean)

Here's an example implementation using the reproduction steps above:

```
"sources": [{
    "source": "https://rubygems.org",
    "enabled": false
  },
  {
    "source":"https://gems.hashicorp.org",
    "enabled":false
  },
  {
    "source":"http://localhost:8081/repository/rubygems.org"
    "enabled":true
  }
]
```

This could be implemented via the cli by adding a new `plugin manage` subcommand and options:

`--disable-default-sources` - Disables the default vagrant plugin sources (rubygems.org and gems.hashicorp.org
`--add-source` - Creates a new global plugin source

Alternatively passing these options without one of the existing plugin subcommands would manage the global plugin behavior (not needing a `manage` subcommand for the `plugin` subcommand)

**Comment by LongLiveCHIEF** [comment]
The documentation for `--plugin-source` reads as follows:

> --plugin-source SOURCE - Adds a source from which to fetch a plugin. Note that this does not only affect the single plugin being installed, by all future plugin as well. This is a limitation of the underlying plugin installer Vagrant uses.

This led me to believe that i could install my first plugin with the `--plugin-source` command, and it would be global, and all future plugins I wouldn't need to define that option.

However, it seems that this option is only applied to "future plugin" of _that specific plugin_.

## steps to reproduce

1. Install vagrant on a clean OS
2. run `vagrant plugin expunge` (just to be sure)
3. Follow the Repro steps in OP to set up rubygems mirror
4. run `vagrant plugin install --plugin-source http://localhost:8081/repository/rubygems.org vagrant-vbguest`
5. Inspect the created `plugins.json` file, and notice that the `sources` key is a member of the `vagrant-proxyconf` object
6. Run `vagrant plugin install vagrant-puppt-install`
7. inspect the `plugins.json` file, notice that the `vagrant-puppet-install` plugin _does not_ use the previously defined plugin source

The documentation is _very_ incorrect here, and at least to the way I am reading it, this feature was never actually implemented the way it was intended.

Right now, it's a single-use command, and only works as intended if the default gem sources are _both_ unreachable.

**Comment by lmayorga1980** [comment]
👍

**Comment by chrisroberts** [comment]
Fixed via #8273. Thanks for the report!
